src/inputs/Button/Button.tsx

```diff
@@ -1,44 +1,52 @@
-import { PropsWithChildren } from "react";
-import * as React from "react";
-
+import React, { PropsWithChildren } from "react";
 import styles from "./Button.module.css";
+import { Spinner } from "../../icons";
+import { Fragment } from "react";
+import classNames from "classnames/bind";
+const cx = classNames.bind(styles);
 
-export interface ButtonProps {
-  onClick(): void;
+interface ButtonProps {
+  type?: "button" | "submit";
   disabled?: boolean;
-  variant?: "primary" | "secondary";
-  isLoading?: boolean;
+  isLoading: boolean;
+  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  variant: "primary" | "secondary";
+  icon: React.ReactNode;
+  children: React.ReactNode;
 }
 
 export const Button = ({
+  type = "button",
   disabled = false,
-  variant = "primary",
-  ...props
+  isLoading,
+  variant,
+  icon,
+  children,
 }: PropsWithChildren<ButtonProps>) => {
-  const variantClass =
-    variant === "primary" ? styles.primary : styles.secondary;
-
-  const getDisabledClassForSpecificVariant = (): string | void => {
-    if (disabled) {
-      return variant === "primary"
-        ? styles.disabledPrimary
-        : styles.disabledSecondary;
-    }
-  };
-
-  const dynamicClasses = [
-    styles.template,
-    variantClass,
-    getDisabledClassForSpecificVariant(),
-  ].join(" ");
+  const buttonClasses = cx({
+    buttons: true,
+    primary: variant === "primary",
+    secondary: variant === "secondary",
+    disabled: disabled,
+    isLoadingPrimary: variant === "primary" && isLoading && !disabled,
```

Comment:
**nitpick:** You duplicate these variant values - it's worth to move the to some `const` or `enum`, like:

`
enum ButtonVariant {
 primary = 'primary'
 ...
}
`